### PR TITLE
Make default sync timeout configurable

### PR DESCRIPTION
**Pull request by BillCarsonFr**
and reduce for tests

 
## Type of change

- [x] Technical:

## Content

Make default sync timeout configurable as part as the MatrixConfiguration in order to reduce it for tests and speed up execution

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [x] Changes has been tested on an Android device or Android emulator with API 21
- [x] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixConfiguration.kt [hunk 1]
```diff
@@ -65,6 +65,12 @@ data class MatrixConfiguration(
          * Thread messages default enable/disabled value
          */
         val threadMessagesEnabledDefault: Boolean = false,
+
+        /**
+         * Sync loop parameter: The maximum time to wait, in milliseconds, before returning this request.
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixConfiguration.kt:70]: This is the server which will wait. Maybe rephrase to make this clearer.

**Review by bmarty — COMMENTED** [review]
Some feedbacks, else LGTM

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixConfiguration.kt [hunk 2]
```diff
@@ -65,6 +65,12 @@ data class MatrixConfiguration(
...
+         * If no events (or other data) become available before this time elapses, the server will return a response with empty fields.
+         */
+        val defaultLongPoolTimeout: Long = 30_000L,
```
[the remark this answers is not in the document]
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixConfiguration.kt]: done

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixConfiguration.kt (hunk 1 above)
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixConfiguration.kt]: updated

**Review by bmarty — APPROVED** [review]
Thanks for the update!

**Review by onurays — APPROVED** [review]
Nice improvement, LGTM!

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/CommonTestHelper.kt [hunk 3]
```diff
@@ -68,7 +68,8 @@ class CommonTestHelper(context: Context) {
                     context,
                     MatrixConfiguration(
                             applicationFlavor = "TestFlavor",
-                            roomDisplayNameFallbackProvider = TestRoomDisplayNameFallbackProvider()
+                            roomDisplayNameFallbackProvider = TestRoomDisplayNameFallbackProvider(),
+                            syncRequestTimeoutMillis = 5_000L
```
Comment (bmarty) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/CommonTestHelper.kt:72]: OOI I am not sure to understand why it should speed up the tests, since if there is something new server side, the server will tell the client immediately, no?

**Review by fedrunov — APPROVED** [review]
LGTM

**Comment by bmarty** [comment]
Superseeded by https://github.com/vector-im/element-android/pull/7198
